Remove crypto

enyo opened the pull request.

sgjesse commented:
Thanks for the patch. However I am a bit concerned about the this code

```
BASE64.encode(_bytes).replaceAll("/", "_").replaceAll("+", "-").replaceAll("=", "");
```

It will create three copies of a potentially long string. In the up-coming Dart 1.16 the BASE64URL codec has been added (https://github.com/dart-lang/sdk/commit/3f0ad9d4f27c0fde6da86f3f0c003ab5bb5184a6), which will get rid of the replaceAll calls.

Also another question, why did you add the final .replaceAll("=", "") to remove the padding chars? The implementation in the crypto package did add the padding chard when `urlSafe` was `true`.

enyo commented:
Yes, I have seen that `urlSafe` has been added to Dart 1.16, but I thought that this would increase the dart dependency from dart 1.13 to 1.16 which would make it much more difficult for many people to upgrade. So the idea was to add it like this, until dart 1.16 has been around a while, then it could be removed.
The reason I removed the `=` signs in the end is that they could potentially also cause problems in an URL and they are not strictly necessary (since it's only padding). But I'm not quite sure how to handle that use case. I hadn't removed them initially but have seen that other implementations also removed them, that's why I decided to do it as well. But I can remove that line if you want.

As for the efficiency, I agree, but I haven't found a better way to do it, other than a regular expression, and I was unsure if that would be more efficient. I can adjust it if you could tell me the most efficient way to do it.
I didn't give it much thought to be honest, since I thought that this will be removed as soon as dart 1.16 has landed.

enyo commented:
I reverted the base64 change that removes the `=` signs and also committed the [googleapis](https://github.com/dart-lang/googleapis/pull/8) again, with the newly compiled apis.

kevmoo commented:
@enyo Could you "squash" your pull request to only one commit?

sgjesse commented:
The SDK dev 1.16.0-dev.5.0 (https://github.com/dart-lang/sdk/commit/e2d59e7f3770ef0cc66a46a0d4f1ec618a3327c0) with urlSafe support is going out today, so I think we should jump directly to that. Adding the sdk constraint for that dev version here and all the way through, so we don't have a transition with a possible hidden performance degradation. People on older SDK will get the older versions of the packages.

enyo commented:
@sgjesse Ok. I'll adapt the PR then.

enyo commented:
@kevmoo GitHub has a squash merge feature now. Should I still do it manually?

sgjesse commented:
@kevmoo we should move to 1.16 where `urlSafe` is supported. @enyo where you preparing that change?

kevmoo commented:
@sgjesse I just sent a pull request that makes sure we're good for 1.15 – doing a 1.16 patch (once we release) is a good idea, too.

enyo commented:
Haven't yet but I can create a PR do we are ready to go.
